hypertrace-metrics-generator/hypertrace-metrics-generator-api/src/main/avro/metric.avdl

```diff
@@ -0,0 +1,9 @@
+@namespace("org.hypertrace.metrics.generator.api")
+protocol MetricProtocol {
+	record Metric {
+		union {null, string} name = null;
```

Comment:
is this different than the key? If we're emitting these for every timestamp, wouldn't we want to put as little in the value message as possible - metadata like name description and unit could be stored elsewhere.

Are attributes holding the numbers? If so, why are they typed string?

Reply:
> is this different than the key?
From this <metric_name> and <attributes> are used form a key. 

So, for a given timestamp window, I just store `Metric` once, and when we emit the metric, that's the time it is used. Reaming, time for each incoming message, to save serde cost, I used `metric_key` for aggregating value in a different store.

> Are attributes holding the numbers? If so, why are they typed string?
This represents the Prometheus metric name and its labels - https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

And, they are strings.

Reply:
> So, for a given timestamp window, I just store Metric once, and when we emit the metric, that's the time it is used. Reaming, time for each incoming message, to save serde cost, I used metric_key for aggregating value in a different store.

Here I meant, that we're storing the attributes and name in every stored metric timewindow, as I understand it. Is that to support querying by attribute and name, not just the id as I was asking about in the other comment thread?

> This represents the Prometheus metric name and its labels - https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels

OK that makes sense, that's what I expected - but where is the actual metric value itself in this message? or it elsewhere? The only numeric-typed thing I see is the timestamp.

Reply:
As part of this PR, I am aggregating `num_calls` metrics to give a time window (say 15s). During this aggregation, I am storing the intermediate result in the state store <MetricIdentity, Value> while punctuating, I am emitting this value after converting to otlp metrics.

- https://github.com/hypertrace/hypertrace-ingester/blob/557750397abdad2f07154a27951d240e56d2837f/hypertrace-metrics-generator/hypertrace-metrics-generator/src/main/java/org/hypertrace/metrics/generator/MetricEmitPunctuator.java#L71
- https://github.com/hypertrace/hypertrace-ingester/blob/557750397abdad2f07154a27951d240e56d2837f/hypertrace-metrics-generator/hypertrace-metrics-generator/src/main/java/org/hypertrace/metrics/generator/MetricsProcessor.java#L120